### ISSUE TITLE
[BUG] kube-api loop times out

### ISSUE THREAD

**Issue by hupiper** [code-ready/crc#2160]
### General information

  * OS: Fedora 33 Kernel 5.11.10-200.fc33.x86_64
  * Hypervisor: KVM 
  * Did you run `crc setup` before starting it (Yes/No)? Yes
  * Running CRC on: Laptop 

## CRC version
```bash
# Put `crc version` output here
CodeReady Containers version: 1.24.0+5f06e84b
OpenShift version: 4.7.2 (embedded in executable)
```
  
## CRC status
```bash
# Put `crc status --log-level debug` output here
$ crc status --log-level debug
DEBU CodeReady Containers version: 1.24.0+5f06e84b 
DEBU OpenShift version: 4.7.2 (embedded in executable) 
DEBU Running 'crc status'                         
DEBU Checking file: /home/hupiper/.crc/machines/crc/.crc-exist 
DEBU Checking file: /home/hupiper/.crc/machines/crc/.crc-exist 
DEBU Found binary path at /home/hupiper/.crc/bin/crc-driver-libvirt 
DEBU Launching plugin server for driver libvirt   
DEBU Plugin server listening at address 127.0.0.1:37627 
DEBU () Calling .GetVersion                       
DEBU Using API Version 1                          
DEBU () Calling .SetConfigRaw                     
DEBU () Calling .GetMachineName                   
DEBU (crc) Calling .GetState                      
DEBU (crc) DBG | time="2021-03-28T13:28:31-05:00" level=debug msg="Getting current state..." 
DEBU (crc) DBG | time="2021-03-28T13:28:31-05:00" level=debug msg="Fetching VM..." 
DEBU (crc) Calling .GetBundleName                 
DEBU (crc) Calling .GetIP                         
DEBU (crc) DBG | time="2021-03-28T13:28:31-05:00" level=debug msg="GetIP called for crc" 
DEBU (crc) DBG | time="2021-03-28T13:28:31-05:00" level=debug msg="Getting current state..." 
DEBU (crc) DBG | time="2021-03-28T13:28:31-05:00" level=debug msg="IP address: 192.168.130.11" 
DEBU Running SSH command: df -B1 --output=size,used,target /sysroot | tail -1 
DEBU Using ssh private keys: [/home/hupiper/.crc/machines/crc/id_ecdsa /home/hupiper/.crc/cache/crc_libvirt_4.7.2/id_ecdsa_crc] 
DEBU SSH command results: err: <nil>, output: 32737570816 10457907200 /sysroot 
DEBU Running SSH command: <hidden>                
DEBU SSH command failed                           
DEBU cannot get OpenShift status: ssh command error:
command : timeout 5s oc get co -ojson --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 124\n 
DEBU Making call to close driver server           
DEBU (crc) Calling .Close                         
DEBU Successfully made call to close driver server 
DEBU Making call to close connection to plugin binary 
DEBU (crc) DBG | time="2021-03-28T13:28:38-05:00" level=debug msg="Closing plugin on server side" 
CRC VM:          Running
OpenShift:       Unreachable (v4.7.2)
Disk Usage:      10.46GB of 32.74GB (Inside the CRC VM)
Cache Usage:     13.17GB
Cache Directory: /home/hupiper/.crc/cache
```

## CRC config
```bash
# Put `crc config view` output here
crc config view
- consent-telemetry                     : yes
```

## Host Operating System
```bash
# Put the output of `cat /etc/os-release` in case of Linux
cat /etc/os-release
NAME=Fedora
VERSION="33 (Workstation Edition)"
ID=fedora
VERSION_ID=33
VERSION_CODENAME=""
PLATFORM_ID="platform:f33"
PRETTY_NAME="Fedora 33 (Workstation Edition)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:33"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f33/system-administrators-guide/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=33
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=33
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Workstation Edition"
VARIANT_ID=workstation

```

### Steps to reproduce

  1. crc setup
  2. crc 
  3. crc start --memory 24576 --log-level debug
  4. 

### Expected
finish and give me the login credentials

### Actual
```
DEBU error: Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 124\n - sleeping 1s 
DEBU retry loop: attempt 130                      
DEBU Running SSH command: timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig 
DEBU SSH command results: err: Process exited with status 124, output:  
DEBU                                              
DEBU error: Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 124\n - sleeping 1s 
DEBU RetryAfter timeout after 131 tries           
DEBU Making call to close driver server           
DEBU (crc) Calling .Close                         
DEBU Successfully made call to close driver server 
DEBU Making call to close connection to plugin binary 
DEBU (crc) DBG | time="2021-03-28T13:26:22-05:00" level=debug msg="Closing plugin on server side" 
Error waiting for apiserver: Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 1\n (x83)
Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 124\n (x3)
Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 1\n (x32)
Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 124\n (x13)
```

### Logs

Before gather the logs try following if that fix your issue
```bash
$ crc delete -f
$ crc cleanup
$ crc setup
$ crc start --log-level debug
```

Please consider posting the output of `crc start --log-level debug`  on http://gist.github.com/ and post the link in the issue.

https://gist.github.com/hupiper/a435d8601a90bbc1d6875d18bc050304

**Comment by cfergeau** [comment]
Could you add the full output of `crc setup --log-level debug` right after `crc delete -f;   crc cleanup; crc setup` ?

**Comment by praveenkumar** [comment]
@hupiper This looks like API server doesn't start at all, can you provide full output what mentioned https://github.com/code-ready/crc/issues/2160#issuecomment-809270014 here?

**Comment by hupiper** [comment]
Yep, I'll post them later this afternoon.

**Comment by hupiper** [comment]
https://gist.github.com/hupiper/c32686ca5be8e65e4adb94bd43a732fb

**Comment by hupiper** [comment]
I provided the data when it was asked for. If you aren't going to look at it, you can close.

**Comment by cfergeau** [comment]
Nothing stands out in that log unfortunately :( I don't know if you've tried newer versions, and are still seeing the same issue?

**Comment by mwarad** [comment]
I face the same issue, let me know if any logs required. I tried with latest version available as well.
INFO Waiting for kube-apiserver availability... [takes around 2min] ************Error waiting for apiserver: Temporary error: ssh command error:****************command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig********************************************************************err     : Process exited with status 1\n (x7)***********************************Temporary error: ssh command error:*********************************************command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig*************
err     : Process exited with status 124\n (x33)
[ansible@manemata ~]$ ^C
[ansible@manemata ~]$ crc status
CRC VM:          Running
OpenShift:       Unreachable (v4.7.11)
Disk Usage:      12.83GB of 32.74GB (Inside the CRC VM)
Cache Usage:     13.32GB
Cache Directory: /home/ansible/.crc/cache

**Comment by samdulam** [comment]
Same error on Fedora 34
```
DEBU error: Temporary error: ssh command error:

command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig

err     : Process exited with status 124\n - sleeping 1s 

DEBU retry loop: attempt 36                       

DEBU Running SSH command: timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig 

DEBU SSH command results: err: Process exited with status 124, output:  

DEBU                                              

DEBU error: Temporary error: ssh command error:

command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig

err     : Process exited with status 124\n - sleeping 1s 

DEBU RetryAfter timeout after 37 tries            

DEBU Making call to close driver server           

DEBU (crc) Calling .Close                         

DEBU Successfully made call to close driver server 

DEBU Making call to close connection to plugin binary 

DEBU Making call to close driver server           

DEBU (crc) Calling .Close                         

DEBU Successfully made call to close driver server 

DEBU Making call to close connection to plugin binary 

Error waiting for apiserver: Temporary error: ssh command error:

command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig

err     : Process exited with status 124\n

Temporary error: ssh command error:

command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig

err     : Process exited with status 1\n (x12)

Temporary error: ssh command error:

command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig

err     : Process exited with status 124\n (x3)

Temporary error: ssh command error:

command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig

err     : Process exited with status 1\n (x7)

Temporary error: ssh command error:

command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig

err     : Process exited with status 124\n (x14)
```

**Comment by guillaumerose** [comment]
Hello,

These logs don't tell us a lot :/ 
Can you try to enter the VM and see what going on inside ? Here is a guide to debug: https://github.com/code-ready/crc/wiki/Debugging-guide
You can check logs of kubelet and see if some containers are scheduled with `crictl ps`. 

Thanks

**Comment by kgjseanbilbo** [comment]
Hi again.
I have been test CRC 129 with CentOS Linux release 8.4.2105

cat /etc/redhat-release 
CentOS Linux release 8.4.2105

And the same result:


time="2021-07-16T06:34:32-04:00" level=debug
time="2021-07-16T06:34:32-04:00" level=debug msg="error: Temporary error: ssh command error:\ncommand : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig\nerr     : Process exited with status 124\\n - sleeping 1s"
time="2021-07-16T06:34:33-04:00" level=debug msg="retry loop: attempt 16"
time="2021-07-16T06:34:33-04:00" level=debug msg="Running SSH command: timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig"
time="2021-07-16T06:34:38-04:00" level=debug msg="SSH command results: err: Process exited with status 124, output: "
time="2021-07-16T06:34:38-04:00" level=debug



Error waiting for apiserver: Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 1\n (x4)
Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 124\n (x37)



Thanks

**Comment by angusfz** [comment]
Same issue here

```
$ cat /etc/system-release
CentOS Linux release 7.9.2009 (Core)
```

detail log https://gist.github.com/angusfz/e83605b1f3b30bdffccfef365d1a7ee9#file-gistfile1-txt

**Comment by cfergeau** [comment]
It looks like the apiserver is not running at all when this fails?
Is this still happening with 4.8-based releases?
Could you try `~/.crc/bin/oc/oc --kubeconfig ~/.crc/cache/crc_libvirt_4.7.18/kubeconfig get co` (replacing `4.7.18` with the OpenShift version if that's a different one)?
Could you also ssh into the node and check the kubelet node?

**Comment by safiyat** [comment]
I am facing the same issue.

When running `crc --log-level=debug start`, I get the following output:

```
...
INFO Starting OpenShift kubelet service
DEBU Using root access: Executing systemctl daemon-reload command
DEBU Running SSH command: sudo systemctl daemon-reload
DEBU SSH command results: err: <nil>, output:
DEBU Using root access: Executing systemctl start kubelet
DEBU Running SSH command: sudo systemctl start kubelet
DEBU SSH command results: err: <nil>, output:
INFO Waiting for kube-apiserver availability... [takes around 2min]
DEBU retry loop: attempt 0
DEBU Running SSH command: timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
DEBU SSH command results: err: Process exited with status 124, output:
DEBU
DEBU error: Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 124
 - sleeping 1s
DEBU retry loop: attempt 1
DEBU Running SSH command: timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
DEBU SSH command results: err: Process exited with status 124, output:
DEBU
DEBU error: Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 124
 - sleeping 1s
DEBU retry loop: attempt 2
DEBU Running SSH command: timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
DEBU SSH command results: err: Process exited with status 124, output:
DEBU
DEBU error: Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 124
 - sleeping 1s

...

DEBU retry loop: attempt 31
DEBU Running SSH command: timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
DEBU SSH command results: err: Process exited with status 124, output:
DEBU
DEBU error: Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 124
 - sleeping 1s
DEBU RetryAfter timeout after 32 tries
DEBU Making call to close driver server
DEBU (crc) Calling .Close
DEBU (crc) DBG | time="2021-12-28T07:55:24Z" level=debug msg="Closing plugin on server side"
DEBU Successfully made call to close driver server
DEBU Making call to close connection to plugin binary
DEBU Making call to close driver server
DEBU (crc) Calling .Close
DEBU Successfully made call to close driver server
DEBU Making call to close connection to plugin binary
DEBU (crc) DBG | time="2021-12-28T07:55:24Z" level=debug msg="Closing plugin on server side"
Error waiting for apiserver:
Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 1

...

Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 124
 (x4)
```

But when I SSH to the VM myself and check after a short while, API server is running.

```
$ ssh -i /home/ubuntu/.crc/machines/crc/id_ecdsa -p 2222 core@127.0.0.1
Red Hat Enterprise Linux CoreOS 49.84.202111292103-0
  Part of OpenShift 4.9, RHCOS is a Kubernetes native operating system
  managed by the Machine Config Operator (`clusteroperator/machine-config`).

WARNING: Direct SSH access to machines is not recommended; instead,
make configuration changes via `machineconfig` objects:
  https://docs.openshift.com/container-platform/4.9/architecture/architecture-rhcos.html

---
[core@crc-hsl9k-master-0 ~]$ oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
NAME                 STATUS   ROLES           AGE   VERSION
crc-hsl9k-master-0   Ready    master,worker   17d   v1.22.3+ffbb954
```

Is there a way to increase the number of retries (or the timeout) via the crc config file?

Also, are there more steps that were not carried out because of the timeout, or is my crc deployment ready?

**Comment by patrykbe777** [comment]
HI, 

The same can be seen on I7-12600K with newer release....

I have assigned much more resources and in the past it worked on i7-8550U ..... what has happen to this software? Version for M1 processors also is not working.

INFO Waiting for kube-apiserver availability... [takes around 2min]
Error waiting for apiserver: Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 1
 (x8)
Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 124
 (x22)
Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 1
 (x2)
Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 124

Patryk

**Comment by erstert3st** [comment]
did you find an Solution ?

**Comment by cfergeau** [comment]
If you are having this issue with recent crc version, can you open a new issue and provide the information asked in the issue template?
